src/Qir/Tools/Driver/QirCppDriverGenerator.cs

```diff
@@ -33,20 +33,15 @@ public async Task GenerateAsync(EntryPointOperation entryPointOperation, Stream
 
         public string GetCommandLineArguments(ExecutionInformation executionInformation)
         {
-            // Sort arguments by position.
-            var sortedArguments = executionInformation.EntryPoint.Parameters.OrderBy(arg => arg.Position);
-            var argumentBuilder = new StringBuilder();
-            foreach (var arg in sortedArguments)
+            string Argument(Parameter param)
             {
-                if (argumentBuilder.Length != 0)
-                {
-                    argumentBuilder.Append(' ');
-                }
-
-                argumentBuilder.Append($"--{arg.Name}").Append(' ').Append(GetArgumentValueString(arg, executionInformation.ArgumentValues[arg.Name]));
+                var prefix = param.Name.Length > 1 ? "--" : "-";
```

Comment:
Is `param.Name.Length == 0` possible?

Reply:
No. (I wish a type like `NonEmptyString` was more common.)